### PR TITLE
Update per review feedback from LLVM

### PR DESCRIPTION
**Pull request by AndyAyersMS**

### CODE REVIEW

**Comment by AndyAyersMS** [comment]
Looks like questionable failures...

@dotnet-bot retest this please

**Comment by AndyAyersMS** [comment]
@dotnet-bot retest this please

**Comment by AndyAyersMS** [comment]
I think the change is fine, but the test that is timing out (which I've tried tweaking to make it run faster) is exactly the one that stresses this change. So will hold off until I can get the test issues cleared up.

**Comment by AndyAyersMS** [comment]
Times looking more reasonable, so will give this one more shot.

@dotnet-bot retest this please

**Comment by AndyAyersMS** [comment]
Ooops, need to rebase...

**Comment by AndyAyersMS** [comment]
@JosephTremoulet PTAL -- incorporate feedback from rnk

File: lib/Target/X86/X86FrameLowering.cpp [hunk 1]
```diff
@@ -450,29 +450,25 @@ MachineInstr *X86FrameLowering::emitStackProbe(MachineFunction &MF,
...
+    if (MI.isCall() && MI.getOperand(0).isSymbol() &&
+        ChkStkStubSymbol == MI.getOperand(0).getSymbolName()) {
+      ChkStkStub = &MI;
+      break;
     }
   }
 
   if (ChkStkStub != nullptr) {
+    MachineBasicBlock::iterator MBBI = ChkStkStub->getIterator();
+    ++MBBI;
```
Comment (JosephTremoulet) [inline lib/Target/X86/X86FrameLowering.cpp]: Style nit: I like `MachineBasicBlock::iterator MBBI = std::next(ChkStkStub->getIterator());` for situations like this (simply for readability, to embed the throwaway intermediate state in the init expression).

**Comment by JosephTremoulet** [comment]
LGTM.  I'm a bit confused about the workflow, though; I had to pull up [D13361](http://reviews.llvm.org/D13361) to get the context, and that review's not closed so possibly more changes could be coming.  I may be missing context, but to me it feels like we should just be having this discussion on that thread and picking up the resulting change from the FI.  If the concern was about making sure parties interested in LLILC have visibility into related changes that go into LLVM, that's something I've been pondering too.  Right now it's just on all of us to watch for interesting reviews (I've recently set up some Herald rules to help me with that).  Maybe we should get in the habit of dropping a pointer in the LLILC gitter whenever we introduce a relevant LLVM review?  And also including LLILC folks as reviewers on such changes (e.g. I tagged @pgavlin on [D14344](http://reviews.llvm.org/D14344))?

(to be clear, this change LGTM and I'm fine with checking it in here now; it just raised some questions in my mind about what we want our typical workflow to be going forward)

**Comment by AndyAyersMS** [comment]
Agreed this is all worth some thought. 

In this case -- I can't test this change without first ensuring that it works with LLILC, and I don't like to put anything up for review in LLVM that we haven't already been using ourselves first. So as clunky as it may seem I feel like I kind of have to iterate back and forth.

**Comment by pgavlin** [comment]
Looks like I need to fix some email filters: I'm no longer getting emails when I'm tagged in LLVM CRs.

FWIW, I agree with Joe that we should be submitting changes upstream and taking them via an FI rather than making changes here, as doing so reduces merge pain. Ideally we'd have enough testing for these changes upstream that we could reliably check in there first.

**Pull request merged**